src/frontends/paddle/src/op/box_coder.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+
+#include "default_opset.hpp"
+#include "openvino/frontend/paddle/node_context.hpp"
+
+namespace ov {
+namespace frontend {
+namespace paddle {
+namespace op {
+NamedOutputs box_coder(const NodeContext& node) {
+    auto prior_box = node.get_input("PriorBox");
+    auto target_box = node.get_input("TargetBox");
+    auto axis = node.get_attribute<int32_t>("axis", 0);
+    auto norm = node.get_attribute<bool>("box_normalized", true);
+    auto uns_axes = default_opset::Constant::create(ov::element::i64, {1}, {axis});
+
+    Output<Node> variance;
+    if (node.has_input("PriorBoxVar")) {
+        variance = node.get_input("PriorBoxVar");
+        variance = std::make_shared<default_opset::Unsqueeze>(variance, uns_axes);
+    }
+    else {
+        std::vector<float> var_vector = node.get_attribute<std::vector<float>>("variance", {1.0, 1.0, 1.0, 1.0});
+        variance = default_opset::Constant::create(ov::element::f32, {4}, var_vector);
+        variance =
+            std::make_shared<default_opset::Unsqueeze>(variance,
+                                                       default_opset::Constant::create(ov::element::i32, {1}, {0}));
+    }
+    auto code_type = node.get_attribute<std::string>("code_type");
+    PADDLE_OP_CHECK(node, (code_type == "decode_center_size"), "Current only supported decode mode!");
+
+    auto target_shape = std::make_shared<default_opset::ShapeOf>(target_box);
+    prior_box = std::make_shared<default_opset::Unsqueeze>(prior_box, uns_axes);
+    prior_box = std::make_shared<default_opset::Broadcast>(prior_box, target_shape);
+
+    // split inputs into 4 elements
+    auto split_axes = default_opset::Constant::create(element::i64, Shape{}, {2});
+    auto prior_split = std::make_shared<default_opset::Split>(prior_box, split_axes, 4);  // pxmin, pymin, pxmax, pymax
+    auto target_split = std::make_shared<default_opset::Split>(target_box, split_axes, 4);  // tx, ty, tw, th
+    split_axes = default_opset::Constant::create(element::i64, Shape{}, {-1});
+    auto var_split = std::make_shared<default_opset::Split>(variance, split_axes, 4);  // pxv, pyv, pwv, phv
+    // convert prior box from [xmin, ymin, xmax, ymax] to [x, y, w, h]
+    OutputVector Priors(4), Target(4);
```

Comment:
Please align the naming conventions. Capitalized names are usually used by classes instead of variables.

Reply:
down